src/frontend/data/stores/useParticipantsStore/index.ts

```diff
@@ -0,0 +1,50 @@
+import create from 'zustand';
+
+export type ParticipantType = {
+  isInstructor: boolean;
+  isOnStage: boolean;
+  name: string;
+};
+
+type State = {
+  addParticipant: (participant: ParticipantType) => void;
+  participants: ParticipantType[];
+  removeParticipant: (participantName: string) => void;
+  updateParticipant: (participant: ParticipantType) => void;
+};
+
+export const useParticipantsStore = create<State>((set) => ({
+  participants: [],
+  addParticipant: (participant) =>
+    set((state) => {
+      if (!state.participants.some((p) => p.name === participant.name)) {
```

Comment:
If the `name` is a unique key in your array, then maybe you should change the `participants` structure to an object and the keys are the participant name ? Doing this you don't have to check if it exists or not. You simply add it, if it exists it will be replaced, if it doesn't exists it will be added. Doing this, the update is useless and the remove will be easier and more performant.